standard/variables.md

```diff
@@ -901,3 +901,181 @@ variable_reference
+
+For an operator that yields a variable, such as `c ? ref e1 : ref e2`:
+
+- The *ref-safe-scope* of the result is the narrowest scope among the *ref-safe-scopes* of the operands of the operator.
+- The *safe-scope* of the operands must agree, and that is the *safe-scope* of the resulting variable.
+
+### §ref-span-safety-method-invocation Method invocation
+
+For a variable `c` resulting from a ref-returning method invocation, `ref e1.M(e2, ...)`, its *ref-safe-scope* is the narrowest of the following scopes:
+
+- The *caller-scope*.
+- The *ref-safe-scope* of all `ref` and `out` argument expressions (excluding the receiver).
+- For each `in` parameter of the method, if there is a corresponding expression that is an variable, its *ref-safe-scope*, otherwise the nearest enclosing scope
+- The *safe-scope* of all argument expressions (including the receiver)
+
+> *Example*: the last bullet is necessary to handle code such as
+>
+> ```csharp
+> var sp = new Span(...)
+> return ref sp[0];
+> ```
+>
+> or
+>
+> ```csharp
+> return ref M(sp, 0);
+> ```
+>
+> *end example*
+
+For a value resulting from a method invocation `e1.M(e2, ...)`, its *safe-scope* from the narrowest of the following scopes:
+
+- The *caller-scope*
+- The *safe-scope* of all argument expressions (including the receiver)
+
+A property invocation (either `get` or `set`) is treated as a method invocation of the underlying method by the above rules.
+
+### §ref-span-safety-a-value Values
+
+A value's *ref-safe-scope* is the nearest enclosing scope.
+
+> *Note:* This occurs in an invocation such as `M(ref d.Length)` where `d` is of type `dynamic`. It is also consistent with arguments corresponding to `in` parameters.
+
+### §ref-span-safety-stackalloc stackalloc
+
+The *safe-scope* for a value `c` resulting from a `stackalloc` expression is the current method.
+
+#### §ref-span-safety-constructor-invocations Constructor invocations
+
+A `new` expression that invokes a constructor obeys the same rules as a method invocation (§ref-span-safety-method-invocation) that is considered to return the type being constructed.
+
+In addition, for a value `c` that is the result of a `new` expression, the *safe-scope* is no wider than the smallest of the *safe-to-escape-scopes* of all arguments and operands of the object initializer expressions, recursively, if any initializer is present.
+
+### §ref-span-safety-default-expressions Default expressions
+
+The *safe-scope* of a `default` expression is the *caller-scope*.
+
+### §ref-span-safety-limitations Limitations on reference variables
+
+- Neither a `ref` parameter, nor a `ref` local, nor a parameter or local of a `ref struct` type can be lifted into a lambda expression or local function.
```

Comment:
Reference for lifting? (Is this the same lifting as in 10.6.2 and 11.4.8?)

Reply:
That's an incorrect term. It should be "captured by"